src/tree/SubscriptionTreeItem.ts

```diff
@@ -113,62 +117,74 @@ export class SubscriptionTreeItem extends SubscriptionTreeItemBase {
         );
     }
 
-    public registerRefreshEvents(key: string): void {
+    public registerRefreshEvents(): void {
+        registerEvent('treeView.onDidChangeFocusedGroup', ext.events.onDidChangeFocusedGroup, async (context: IActionContext) => {
+            context.errorHandling.suppressDisplay = true;
+            context.telemetry.suppressIfSuccessful = true;
+            context.telemetry.properties.isActivationEvent = 'true';
+            await this.refresh(context);
+        });
+
         registerEvent('treeView.onDidChangeConfiguration', workspace.onDidChangeConfiguration, async (context: IActionContext, e: ConfigurationChangeEvent) => {
             context.errorHandling.suppressDisplay = true;
             context.telemetry.suppressIfSuccessful = true;
             context.telemetry.properties.isActivationEvent = 'true';
 
-            if (e.affectsConfiguration(`${ext.prefix}.${key}`)) {
+            if (e.affectsConfiguration(`${ext.prefix}.groupBy`)) {
+                // we can generate the default groups ahead of time so we don't need to refresh the entire tree
+                this._triggeredByDefaultSetting = Object.values(GroupBySettings).includes(settingUtils.getWorkspaceSetting<string>('groupBy') as GroupBySettings);
                 await this.refresh(context);
```

Comment:
Should we also wrap this in a try/finally to make sure we reset `this._triggeredByDefaultSetting`? Especially if we ever implement `refreshImpl`.

Reply:
Well, the thing is, even if you await the refresh, it still technically resolves before loadMoreChildren is actually called apparently.  I originally did try/catch the setting here, but it kept getting set before loadMoreChildren would actually start